src/components/analyser/search/DashboardTable.tsx

```diff
@@ -141,17 +142,46 @@ export const DashboardTableRow: React.FC<DashboardTableRowProps> = ({
   );
 };
 
-export type DashboardTableProps = {
-  stocks: API.Stock[];
-};
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+function sortStocks(
+  items: API.Stock[],
+  order: Order,
+  orderBy: keyof API.Stock
+) {
+  return order === 'desc'
+    ? items.sort((a, b) => descendingComparator(a, b, orderBy))
+    : items.sort((a, b) => -descendingComparator(a, b, orderBy));
+}
 
 const DashboardTable: React.FC<DashboardTableProps> = ({ stocks }) => {
-  const { t } = useTranslation();
   const classes = useStyles();
 
   const [items, setItems] = React.useState<API.Stock[]>(stocks.slice(0, 10));
   const [hasMore, setHasMore] = React.useState<boolean>(true);
 
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderByKey, setOrderByKey] = React.useState<keyof API.Stock>('name');
+  const [sortedStocks, setSortedStocks] = React.useState(() =>
+    sortStocks(items, order, orderByKey)
+  );
```

Comment:
pls put some comment in there to explain what they are for and how it is working